src/base/BaseActionsRouter.sol

```diff
@@ -52,4 +53,17 @@ abstract contract BaseActionsRouter is SafeCallback {
     /// `msg.sender` shouldnt be used, as this will be the v4 pool manager contract that calls `unlockCallback`
     /// If using ReentrancyLock.sol, this function can return _getLocker()
     function _msgSender() internal view virtual returns (address);
+
+    /// @notice Calculates the address for a action
+    /// @param recipient The address or address-flag for the action
+    /// @return output The resultant address for the action
+    function map(address recipient) internal view returns (address) {
```

Comment:
wouldve been nice in a library but sadly it needs `_msgSender` to work. Open to suggestions about where the function should live, and where the constants should live